textpipe/doc.py

```diff
@@ -512,21 +513,48 @@ def aggregate_word_vectors(self, model_name=None, aggregation='mean', normalize=
         else:
             raise NotImplementedError(f'Aggregation method {aggregation} is not implemented.')
 
-    def _load_gensim_word2vec_model(self, model_file=None):
+    def _load_gensim_word2vec_model(self,
+                                    model_file=None,
+                                    redis_host=None,
+                                    redis_port=None,
+                                    redis_db=0,
+                                    max_lru_cache_size=1024):
         """
-        Loads pre-trained Gensim word2vec keyed vector model
+        Loads pre-trained Gensim word2vec keyed vector model from either local or Redis
         >>> model = Doc('')._load_gensim_word2vec_model('tests/models/gensim_test_nl.kv')
         >>> type(model)
         <class 'gensim.models.keyedvectors.Word2VecKeyedVectors'>
         """
         lang = self.language if self.is_reliable_language else self.hint_language
         if not self._gensim_vectors or lang not in self._gensim_vectors:
-            vectors = KeyedVectors.load(model_file, mmap='r')
+            if redis_host and redis_port:
```

Comment:
It might be nicer to have a 'use_redis' flag for this when you want to use the library with a component that uses redis for other things as well.

Reply:
We have to pass these parameters to the textpipe pipeline configuration explicitly for this operation, so I don't think the reason you mention for the redis flag is a valid one. I thought about such a flag, but I couldn't think of a case where it would be `False` when the host and port would have a value or the other way around. It does hence not encode any information.